Sources/SwiftDocC/Infrastructure/DocumentationConverter.swift

```diff
@@ -412,6 +406,34 @@ public struct DocumentationConverter: DocumentationConverterProtocol {
         return isDocumentPathToConvert || isExternalIDToConvert
     }
     
+    /// Record a problem from the given error in the given problem array.
+    ///
+    /// Creates a ``Problem`` from the given `Error` and identifier, emits it to the
+    /// ``DocumentationConverter``'s ``DiagnosticEngine``, and appends it to the given
+    /// problem array.
+    ///
+    /// - Parameters:
+    ///   - error: The error that describes the problem.
+    ///   - problems: The array that the created problem should be appended to.
+    ///   - identifier: A unique identifier the problem.
+    private func recordProblem(
+        from error: Swift.Error,
+        in problems: inout [Problem],
+        withIdentifier identifier: String
```

Comment:
This change just simplified the way we report errors in DocumentationConverter to make it less likely we'll introduce the problem of accidentally including a `source` location in the future.

It also updates the `identifier` used to actually be unique for each item. Currently, we're using the `org.swift.docc.DocumentationConverter` identifier for all of them.